moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp

```diff
@@ -0,0 +1,335 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, PickNik Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <moveit/hybrid_planning_manager/hybrid_planning_manager.h>
+#include <moveit/hybrid_planning_manager/hybrid_planning_events.h>
+
+namespace
+{
+const rclcpp::Logger LOGGER = rclcpp::get_logger("hybrid_planning_manager");
+}
+
+namespace moveit::hybrid_planning
+{
+using namespace std::chrono_literals;
+
+HybridPlanningManager::HybridPlanningManager(const rclcpp::NodeOptions& options)
+  : Node("hybrid_planning_manager", options), initialized_(false)
+{
+  // Initialize hybrid planning component after after construction
+  // TODO(sjahr) Remove once life cycle component nodes are available
+  timer_ = this->create_wall_timer(1ms, [this]() {
+    if (initialized_)
+    {
+      timer_->cancel();
+    }
+    else
+    {
+      if (!this->initialize())
+      {
+        const std::string error = "Failed to initialize global planner";
+        timer_->cancel();
+        throw std::runtime_error(error);
+      }
+      initialized_ = true;
+    }
+  });
```

Comment:
We shouldn't need this.  I will create a PR to remove this ( I thought I did this once before ).

Reply:
I'm closing this as future work.